libevm/register/register.go

```diff
@@ -66,3 +66,28 @@ func (o *AtMostOnce[T]) TestOnlyClear() {
 		o.v = nil
 	})
 }
+
+// TempOverride calls `fn`, overriding any registered `T`, but only for the life
+// of the call.
+//
+// It is valid to call this method with or without a prior call to
+// [AtMostOnce.Register].
+func (o *AtMostOnce[T]) TempOverride(with T, fn func()) {
+	o.temp(&with, fn)
+}
+
+// TempClear calls `fn`, clearing any registered `T`, but only for the life of
+// the call.
+//
+// It is valid to call this method with or without a prior call to
+// [AtMostOnce.Register].
+func (o *AtMostOnce[T]) TempClear(fn func()) {
+	o.temp(nil, fn)
+}
+
+func (o *AtMostOnce[T]) temp(with *T, fn func()) {
+	old := o.v
```

Comment:
do we need to use a lock here in case two routines access this? (or temp access + normal access)

Reply:
I've added the `temporary` package with a global lock. If concurrent workers were to both attempt to lock all overrides themselves then, unless they did it in the exact same order, they could deadlock.